tests/testthat/test-correct_time.R

```diff
@@ -0,0 +1,214 @@
+    suppressMessages(correct_time(x, c("62c200a9", "29b7d356"), duration))
+  )
+  # `deploymentID`contains all the deploymentIDs
+  expect_no_error(
+    suppressMessages(
+      correct_time(
+        x, c("00a2c20d", "29b7d356", "577b543a", "62c200a9"), duration
+      )
+    )
+  )
+})
+
+test_that("correct_time() returns no error and corrects deploymentStart when
+          duration has class Duration", {
+  skip_if_offline()
+  x <- example_dataset()
+  depID <- "00a2c20d"
+
+  # Wrong deploymentStart
+  deploymentStart <-
+    deployments(x) %>%
+    dplyr::filter(.data$deploymentID == depID) %>%
+    dplyr::pull(deploymentStart)
+
+  # Set parameters
+  wrong_Duration <- lubridate::ymd_hms("2024-04-01T00:00:00", tz = "UTC")
+  right_Duration <- lubridate::ymd_hms("2024-04-01T02:00:00", tz = "UTC")
+  duration_Duration <-
+    lubridate::as.duration(lubridate::interval(wrong_Duration, right_Duration))
+
+  # No error test
+  expect_no_error(suppressMessages(correct_time(x, depID, duration_Duration)))
+
+  x_Duration <- suppressMessages(correct_time(x, depID, duration_Duration))
+
+  # New deploymentStart
+  deploymentStart_Duration <-
+    deployments(x_Duration) %>%
+    dplyr::filter(.data$deploymentID == depID) %>%
+    dplyr::pull(deploymentStart)
+
+  # Tests
+  expect_identical(
+    deploymentStart_Duration,
+    deploymentStart + duration_Duration
+    )
+  expect_identical(
+    deploymentStart_Duration,
+    as.POSIXct("2020-05-30 04:57:37", tz = "UTC")
+    )
+})
+
+test_that("correct_time() returns no error and corrects deploymentStart when
+          deploymentID has class difftime", {
+  skip_if_offline()
+  x <- example_dataset()
+  depID <- "00a2c20d"
+
+  # wrong deploymentStart
+  deploymentStart <-
+    deployments(x) %>%
+    dplyr::filter(.data$deploymentID == depID) %>%
+    dplyr::pull(deploymentStart)
+
+  # Option 1 to obtain a difftime object
+  wrong_POSIXct <- as.POSIXct("2024-04-01 00:00:00", tz = "UTC")
+  right_POSIXct <- as.POSIXct("2024-04-01 02:00:00", tz = "UTC")
+  duration_difftime1 <- right_POSIXct - wrong_POSIXct
+
+  # Option 2 to obtain a difftime object
+  wrong <- "2024-04-01 00:00:00"
+  right <- "2024-04-01 02:00:00"
+  duration_difftime2 <- difftime(right, wrong, tz = "UTC")
+
+  # Test if both yield the same result
+  expect_identical(duration_difftime1, duration_difftime2)
+
+  # Test further with only duration_difftime1
+  # No error test
+  expect_no_error(suppressMessages(correct_time(x, depID, duration_difftime1)))
+
+  x_difftime <- suppressMessages(correct_time(x, depID, duration_difftime1))
+
+  # New deploymentStart
+  deploymentStart_difftime <-
+    deployments(x_difftime) %>%
+    dplyr::filter(.data$deploymentID == depID) %>%
+    dplyr::pull(deploymentStart)
+
+  # Tests
+  expect_identical(
+    deploymentStart_difftime,
+    deploymentStart + duration_difftime1
+  )
+  expect_identical(
+    deploymentStart_difftime,
+    as.POSIXct("2020-05-30 04:57:37", tz = "UTC")
+  )
+})
```

Comment:
Same here: media and observations are also changed and so tests should be added.